src/caret_analyze/architecture/architecture.py

```diff
@@ -466,6 +467,82 @@ def rename_topic(self, src: str, dst: str) -> None:
         for c in self._communications:
             c.rename_topic(src, dst)
 
+    @staticmethod
+    def diff_node_names(
+        left_arch: Architecture,
+        right_arch: Architecture
+    ) -> Tuple[Tuple[str, ...], Tuple[str, ...]]:
+        """
+        Compare two architecture objects (: left_arch, right_arch) and returns\
+        the node names that exist only in the respective architectures.
```

Comment:
For all docstring:

The explanation is redundant.

```suggestion
        Compare two architecture objects and return the difference of nodes name.
```